### PR TITLE
#165894987 Build out the user signup endpoint

### PR DESCRIPTION
**Pull request by danose**
#### What does this PR do?
Have user signup endpoint for the Quick credit app

#### Description of Task to be completed?
Have the endpoint below working:
POST /api/v1/loans

#### How should this be manually tested?
- Using Post with localhost of 5000, test the endpoint

#### What are relevant pivotal tracker stories?
#165894987

### CODE REVIEW

File: api/v1/controllers/usersController.js [hunk 1]
```diff
@@ -1,65 +1,61 @@
...
+     * @param {Array} res
+     * @returns {object} user json object
+     * @returns {object} error object
+     */
+  static async signUpUsers(req, res) {
+    try {
+      const user = await UserModel.createUser(req.body);
+      const token = createToken(user.id, user.is_admin);
+      return res.status(201)
+        .header('x-auth-access', token)
```
Comment (V-84) [inline api/v1/controllers/usersController.js]: Remove this.. you don't need to put this in the header

File: api/v1/controllers/usersController.js [hunk 2]
```diff
@@ -1,65 +1,61 @@
...
+        .json({
+          status: 201,
+          data: {
+            token,
+            id: user.id,
+            firstName: user.first_name,
```
Comment (V-84) [inline api/v1/controllers/usersController.js]: Remove the user information and store them in the token instead

**Review by V-84 — CHANGES_REQUESTED** [review]
Attend to the comments

File: api/v1/controllers/usersController.js [hunk 3]
```diff
@@ -1,65 +1,53 @@
...
+        firstName: user.first_name,
+        email: user.email,
+        lastName: user.last_name,
+        address: user.address,
+        phone: user.phone };
+      const token = createToken(user.id, user.is_admin, userInfo);
+      return res.status(201)
+        .json({
+          status: 201,
+          token
```
Comment (V-84) [inline api/v1/controllers/usersController.js:30]: add `message: 'Success'`

File: api/v1/controllers/usersController.js [hunk 4]
```diff
@@ -1,65 +1,53 @@
...
+        });
+    } catch (error) {
+      if (error.routine === '_bt_check_unique') {
+        return res.status(400).json({ status: 400,
+          error: 'user with that EMAIL or phone number already exists'
```
Comment (V-84) [inline api/v1/controllers/usersController.js]: replace `error` with `message`

Comment (V-84) [reply api/v1/controllers/usersController.js]: status code should be 409

File: api/v1/controllers/usersController.js [hunk 5]
```diff
@@ -1,65 +1,53 @@
...
-
-      return res.status(400).json({
-        status: 400,
-        error: 'user with that EMAIL already exists'
-
-      });
-
+  /**
+     * User signup controller
+     * @param {string} req
```
Comment (V-84) [inline api/v1/controllers/usersController.js]: req is an object

File: api/v1/controllers/usersController.js [hunk 6]
```diff
@@ -1,65 +1,53 @@
...
+     * @param {Array} res
```
Comment (V-84) [inline api/v1/controllers/usersController.js]: res is an object

**Review by V-84 — COMMENTED** [review]
attend to the comment

File: api/v1/controllers/usersController.js (hunk 4 above)
Comment (danose) [reply api/v1/controllers/usersController.js]: Noted

File: api/v1/controllers/usersController.js (hunk 5 above)
Comment (danose) [reply api/v1/controllers/usersController.js]: noted

**Pull request merged**